esphome/components/packet_transport/packet_transport.cpp

```diff
@@ -314,11 +314,36 @@ void PacketTransport::send_data_(bool all) {
 }
 
 void PacketTransport::update() {
-  auto now = millis() / 1000;
+  uint32_t now = millis() / 1000u;
   if (this->last_key_time_ + this->ping_pong_recyle_time_ < now) {
     this->resend_ping_key_ = this->ping_pong_enable_;
+    ESP_LOGV(TAG, "Ping request, age %u", now - this->last_key_time_);
     this->last_key_time_ = now;
   }
+#ifdef USE_STATUS_SENSOR
+  for (const auto &provider : this->providers_) {
+    if (provider.second.status_sensor != nullptr) {
+      uint32_t key_response_age = now - provider.second.last_key_response_time;
+      if (key_response_age > (this->ping_pong_recyle_time_ * 2u)) {
+        if (provider.second.status_sensor->state) {
+          ESP_LOGW(TAG, "Ping status for %s timeout at %u with age %u", provider.first.c_str(), now, key_response_age);
+          provider.second.status_sensor->publish_state(false);
+        }
+        for (auto &sensor : this->remote_sensors_[provider.first]) {
+          sensor.second->publish_state(0);
```

Comment:
```suggestion
          sensor.second->publish_state(NAN);
```
You will also need to guard this with `#ifdef USE_SENSOR`

Reply:
Fixed according to comment